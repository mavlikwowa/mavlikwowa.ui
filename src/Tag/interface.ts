export interface TagProps extends React.HTMLProps<HTMLDivElement> {
  background?: string;
  border?: string;
  width?: string;
  height?: string;
  color?: string;
}
