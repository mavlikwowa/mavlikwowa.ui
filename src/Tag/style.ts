/* Libraries */
import styled from '@emotion/styled';
/* Interfaces */
import { TagProps } from './interface';

export const StyledTag = styled.div<TagProps>`
  ${({ background, border, width, height, color }: TagProps) => {
    return `
    display: inline-flex;
    align-items: center;
    justify-content: center;
    background: ${background || 'white'};
    border: 1px solid ${border || 'grey'};
    width: ${width || '100px'};
    height: ${height || '50px'};
    color: ${color || 'black'};
    padding: 0 12px 0 12px;
    cursor: default;
  `;
  }}
`;
