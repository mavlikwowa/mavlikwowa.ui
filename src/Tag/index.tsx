/* Libraries */
import React from 'react';
/* Interfaces */
import { TagProps } from './interface';
/* Styled Components */
import { StyledTag } from './style';

/* Styled components */
const Tag: React.FC<TagProps> = ({
  height,
  width,
  border,
  background,
  color,
  children,
  style,
}: TagProps) => {
  return (
    <StyledTag
      height={height}
      width={width}
      background={background}
      border={border}
      color={color}
      style={style}
    >
      {children}
    </StyledTag>
  );
};

export default Tag;
