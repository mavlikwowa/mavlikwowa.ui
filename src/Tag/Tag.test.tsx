/* Libraries */
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';

/* Components */
import Tag from './index';

/* Interfaces */
import { TagProps } from './interface';

/** Setups */
const setUp = () => {
  /* A test`s data */
  const props: TagProps = {
    color: '#D0FFC9',
    border: '#ADD900',
    background: '#129900',
    width: '110px',
    height: '35px',
    children: 'Tag text',
  };
  /* Component */
  const wrapper = shallow(<Tag {...props}>{props.children}</Tag>);
  return { props, wrapper };
};

/* Tests */
describe('Test component Tag', () => {
  let wrapper: ShallowWrapper;
  let props: TagProps;

  beforeEach(() => {
    ({ wrapper, props } = setUp());
  });

  afterEach(() => {
    if (!wrapper) return;
    wrapper.unmount();
  });

  it('should render a component', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should find a text of button', () => {
    expect(wrapper.text()).toBe(props.children);
  });
});
