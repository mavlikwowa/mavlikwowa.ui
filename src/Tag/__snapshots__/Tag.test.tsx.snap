// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component Tag should render a component 1`] = `
<Styled(div)
  background="#129900"
  border="#ADD900"
  color="#D0FFC9"
  height="35px"
  width="110px"
>
  Tag text
</Styled(div)>
`;
